apps/camera/js/controllers/battery.js

```diff
@@ -41,6 +44,84 @@ BatteryController.prototype.bindEvents = function() {
   bind(this.battery, 'levelchange', this.updateStatus);
   bind(this.battery, 'chargingchange', this.updateStatus);
   this.app.on('change:batteryStatus', this.onStatusChange);
+  this.app.on('camera:configured', this.updateScreenTimeout);
+  this.app.on('camera:previewactive', this.onPreviewActive);
+
+  var mozSettings = navigator.mozSettings;
+  mozSettings.addObserver('powersave.enabled', this.onPowerSaveChange);
+  mozSettings.createLock().get('powersave.enabled').then(
+    this.onPowerSaveChange);
+};
+
+BatteryController.prototype.onPowerSaveChange = function(values) {
+  var value;
+  if (typeof(values.settingValue) !== 'undefined') {
+    value = values.settingValue;
+  } else {
+    value = values['powersave.enabled'];
+  }
+  this.powerSave = value;
+  debug('power save ' + value);
+  this.updateScreenTimeout();
+  this.app.emit('battery:powersave', value);
+};
+
+BatteryController.prototype.onPreviewActive = function(state) {
+  if (!state) {
+    this.restoreScreenTimeout();
+  }
+};
+
+/**
+ * Restore the screen timeout to the original value before
+ * the camera was opened and power save mode was entered.
+ *
+ * @private
+ */
+BatteryController.prototype.restoreScreenTimeout = function() {
+  this.cachingScreenTimeout = false;
+  if (typeof(this.cachedScreenTimeout) === 'undefined') { return; }
+  debug('restore screen timeout %d', this.cachedScreenTimeout);
+  navigator.mozSettings.createLock().set({
+    'screen.timeout': this.cachedScreenTimeout
+  });
+  delete this.cachedScreenTimeout;
+};
+
+/**
+ * If in power save mode, change the screen timeout to the
+ * app-specific screen timeout, otherwise restore it to the
+ * original timeout period.
+ *
+ * @private
+ */
+BatteryController.prototype.updateScreenTimeout = function() {
+  if (this.app.hidden ||
+      typeof(this.powerSaveScreenTimeout) === 'undefined') {
+    return;
+  }
+
+  if (this.powerSave) {
+    if (typeof(this.cachedScreenTimeout) !== 'undefined' ||
+        this.cachingScreenTimeout) { return; }

```

Comment:
if (this.cachedScreenTimeout !== undefined || this.cachingScreenTimeout) { return; }